src/server/service/quoteEvent.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  AnyChannel,
+  Client,
+  Guild,
+  Message,
+  MessageEmbed,
+  TextChannel
+} from 'discord.js';
+
+async function getMessage(client: Client, receiptMsg: Message) {
+  // メッセージの受け取り
+  // BOTとDMからのメッセージは弾く
+  if (receiptMsg.author.bot || !receiptMsg.guild) return;
+
+  // https://(ptb.|canary.)?discord.com/channels/サーバーID/チャンネルID/メッセージID
+  const receiptLink = new RegExp(
+    /https:\/\/(?:ptb.|canary.)?discord(?:app)?.com\/channels\/(\d+)\/(\d+)\/(\d+)/
+  );
+  const str = receiptMsg.content;
+  const match = str.match(receiptLink);
+  if (!match) return;
+  const [, serverId, channelId, messageId] = match;
+  const quoteChannel = await client.channels.fetch(`${channelId}`); // 引用元チャンネル
+  const quoteServer = await client.guilds.fetch(`${serverId}`); // 引用元サーバー
+  if (receiptMsg.content.startsWith(';')) {
+    console.info(`Info: "` + receiptMsg.author.username + `" used quote skip.`);
+    return;
+  }
+  return {
+    serverId,
+    messageId,
+    quoteChannel,
+    quoteServer
+  };
+}
+
+function sanitize(
+  serverId: string,
+  receiptMsg: Message,
+  quoteChannel: AnyChannel | null,
+  quoteServer: Guild
+): asserts quoteChannel is TextChannel {
+  if (serverId !== receiptMsg.guild?.id) {
+    throw new Error(
+      `error: "` +
+        receiptMsg.author.username +
+        `" Cancel quoting a message from another server.`
+    );
+  }
+  if (!quoteChannel || !quoteServer) {
+    throw new Error(
+      'Error: I was unable to discover the quoting channel and the quoting server.'
+    );
+  }
+  if (!quoteChannel?.isText()) {
+    throw new Error(
+      'Error: The channel of the message for which the citation request was issued is not a text channel.'
+    );
+  }
+}
+
+async function getQuote(quoteChannel: TextChannel, messageId: string) {
+  const quoteMessage = await quoteChannel.messages.fetch(messageId);
+
+  if (!quoteMessage) return;
+  if (quoteMessage.system) return;
+
+  return quoteMessage;
+}
+
+function createQuoteEmbed(quoteMessage: Message) {
+  const startTime = performance.now();
+  console.info(`* Quote Create Start >>> Start creating citations.`);
+
+  const quoteUser = quoteMessage.author;
+  const quoteChannelId = quoteMessage.channel.id;
+  const quoteUserAvatar = quoteUser.avatarURL();
+
+  if (!quoteUser) {
+    throw new Error(
+      'The user information of the quoted message could not be retrieved from the Discord API.'
+    );
+  }
+
+  if (!quoteChannelId) {
+    throw new Error(
+      'Error: The channel ID of the quoted message could not be retrieved from the Discord API.'
+    );
+  }
+
+  const quoteCreateTstamp = quoteMessage.createdTimestamp / 1000;
+
+  const sendQuoteEmbed = new MessageEmbed()
+    .setDescription(quoteMessage.content)
+    .setColor('#FFC9E9')
+    .setAuthor({ name: quoteUser.username + `(${quoteUser.id})` })
+    .addField('チャンネル', '<#' + quoteChannelId + '>', true)
```

Comment:
こういうものはテンプレート文字列を使うべきかと
```suggestion
    .setAuthor({ name: `${quoteUser.username}(${quoteUser.id})` })
    .addField('チャンネル', `<#${quoteChannelId}>`, true)
```